### PR TITLE
enh(assistant builder): check if handle is taken

### PR DESCRIPTION
**Pull request by fontanierh**
case insensitive as per https://dust4ai.slack.com/archives/C050A0S2Z7F/p1695141268023229

### CODE REVIEW

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 1]
```diff
@@ -155,10 +159,11 @@ export default function AssistantBuilder({
...
     return /^[a-zA-Z0-9_-]{1,20}$/.test(handle);
   };
-  const assistantHandleIsAvailable = async (handle: string) => {
-    // TODO: check if handle is available
-    void handle;
-    return true;
+  const assistantHandleIsAvailable = (handle: string) => {
+    return !agentConfigurations.some(
+      (agentConfiguration) =>
+        agentConfiguration.name.toLowerCase() === handle.toLocaleLowerCase()
```
Comment (spolu) [inline front/components/assistant_builder/AssistantBuilder.tsx:165]: one or the other?

**Review by spolu — DISMISSED** [review]

File: front/components/assistant_builder/AssistantBuilder.tsx (hunk 1 above)
Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilder.tsx:165]: wdym ?

Comment (spolu) [reply front/components/assistant_builder/AssistantBuilder.tsx:165]: `toLowerCase` vs `toLocaleLowerCase` ?

Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilder.tsx:165]: missed it ! thanks

**Review by spolu — APPROVED** [review]

**Pull request merged**